src/main/scala/org/clulab/wm/eidos/EidosSystem.scala

```diff
@@ -149,26 +143,38 @@ class EidosSystem(val config: Config = EidosSystem.defaultConfig) {
   // MAIN PIPELINE METHOD if given text
   def extractFromText(
     text: String,
-    keepText: Boolean = true,
     cagRelevantOnly: Boolean = true,
-    documentCreationTime: Option[String] = None,
+    dctString: Option[String] = None,
     filename: Option[String] = None): AnnotatedDocument = {
 
-    val eidosDoc = annotate(text, keepText, documentCreationTime, filename)
-    extractFromDoc(eidosDoc, keepText, cagRelevantOnly, documentCreationTime, filename)
+    val document = annotate(text)
+    extractFromDoc(document, cagRelevantOnly, dctString, filename)
   }
 
   // MAIN PIPELINE METHOD if given doc
   def extractFromDoc(
-      doc: EidosDocument,
-      keepText: Boolean = true,
+      doc: Document,
       cagRelevantOnly: Boolean = true,
-      documentCreationTime: Option[String] = None,
+      dctStringOpt: Option[String] = None,
       filename: Option[String] = None): AnnotatedDocument = {
+    // It is assumed and not verified that the document _has_ already been annotated.
+    require(doc.text.isDefined)
+
+    doc.id = filename
+    if (dctStringOpt.isDefined && timeNormFinderOpt.isDefined) {
+      val dctString = dctStringOpt.get
+      val dctOpt = timeNormFinderOpt.get.parseDctString(dctString)
```

Comment:
I would write the above three lines as:
```
for (dctString <- dctStringOpt; timeNormFinder <- timeNormFinderOpt) {
  val dctOpt = timeNormFinder.parseDctString(dctString)
```

Reply:
I generally don't like `for`, but you have convinced me in this case!  I don't know how that could have happened.

Reply:
=) You could also write it using the equivalent method calls. Something like:
```
val dctOpt = dctStringOpt.flatMap(dctString => timeNormFinder.flatMap(_.parseDctString(dctString))
```
Not sure I've got all my maps flatMaps right there, but hopefully you get the idea.